trieste/acquisition/rule.py

```diff
@@ -680,16 +675,14 @@ def acquire(
                 f"""datasets must be provided and contain the single key {OBJECTIVE}"""
             )
 
-        if self._num_fourier_features is None:  # Perform exact Thompson sampling
+        if self._use_fourier_features is False:  # Perform exact Thompson sampling
             thompson_sampler: ThompsonSampler = ExactThompsonSampler(
                 self._num_query_points, models[OBJECTIVE]
             )
         else:  # Perform approximate Thompson sampling
-            thompson_sampler = RandomFourierFeatureThompsonSampler(
+            thompson_sampler = ThompsonSamplerFromTrajectory(
                 self._num_query_points,
                 models[OBJECTIVE],
-                datasets[OBJECTIVE],
-                num_features=self._num_fourier_features,
             )
```

Comment:
I'm a bit confused by this. Why is there a different case for the using Fourier features versus not using Fourier features? Shouldn't we just let the model decide how to get the Thompson samples, since this is discrete Thompson sampling?

Reply:
For instance the current trajectory sampler for the DGP models doesn't use Fourier features at all

Reply:
This isn't new for this PR. But we previously allowed the use of RFF for getting the discrete Thompson samples if you had a big model or wanted a coarse grid of candidate points. This will be refactored properly in the next PR